app/src/main/java/com/github/orkest/View/profile/ProfileTopInterface.kt

```diff
@@ -72,19 +77,51 @@ fun ProfileTopInterface(viewModel: ProfileViewModel) {
 
         Spacer(modifier = Modifier.height(separator))
 
-        Row(){
-            EditButton {
-                val intent = Intent(context, EditProfileActivity::class.java)
-                context.startActivity(intent)
+
+        Row{
+            if(viewModel.username.value == Constants.currentLoggedUser) {
+                EditButton {
+                    val intent = Intent(context, EditProfileActivity::class.java)
+                    context.startActivity(intent)
+                }
+            } else {
+                FollowButton(viewModel, viewModel.isUserFollowed.observeAsState().value)
             }
         }
     }
 }
 
+@Composable
+fun FollowButton(viewModel: ProfileViewModel, isUserFollowed: Boolean?){
+    if(isUserFollowed == null){ Text(text="") } //Empty body, here waiting for the future that fetch isUserFollowed to complete
+    else {
+        Button(

```

Comment:
It would be great to try to refactor this part to improve readability by making the function shorter (by separating the different UI components for example and the onClick listener)

Reply:
I implemented that in my new commit. Thank you!